crates/holochain/src/conductor/interface/websocket.rs

```diff
@@ -165,58 +203,111 @@ async fn recv_incoming_admin_msgs<A: InterfaceApi>(api: A, rx_from_iface: Websoc
 /// polling for signals being broadcast from the Cells associated with this
 /// App interface.
 fn spawn_recv_incoming_msgs_and_outgoing_signals<A: InterfaceApi>(
+    task_list: &mut TaskList,
     api: A,
     rx_from_iface: WebsocketReceiver,
     rx_from_cell: broadcast::Receiver<Signal>,
     tx_to_iface: WebsocketSender,
 ) {
     use futures::stream::StreamExt;
 
-    trace!("CONNECTION: {}", rx_from_iface.remote_addr());
+    trace!("CONNECTION: {}", rx_from_iface.peer_addr());
 
     let rx_from_cell = futures::stream::unfold(rx_from_cell, |mut rx_from_cell| async move {
-        if let Ok(item) = rx_from_cell.recv().await {
-            Some((item, rx_from_cell))
-        } else {
-            None
-        }
-    });
-
-    tokio::task::spawn(rx_from_cell.for_each_concurrent(4096, move |signal| {
-        let mut tx_to_iface = tx_to_iface.clone();
-        async move {
-            trace!(msg = "Sending signal!", ?signal);
-            if let Err(err) = async move {
-                let bytes = SerializedBytes::try_from(signal)?;
-                tx_to_iface.signal(bytes).await?;
-                InterfaceResult::Ok(())
-            }
-            .await
-            {
-                error!(?err, "error emitting signal");
+        loop {
+            match rx_from_cell.recv().await {
+                // We missed some signals, but the channel is still open
+                Err(tokio::sync::broadcast::error::RecvError::Lagged(_)) => {
+                    warn!("SignalChannelOverloaded");
```

Comment:
This is user-facing so I wonder if a hint here might be useful? Something like "{app_id} is sending signals too fast". Just so it's clear what the source of this is and that it's likely an app error, rather than a Holochain issue?

Reply:
Yes please, and including the Lagged count in the message would be helpful to gauge the severity. Other similar warnings like DB and validation saturation have been helpful, and they also print some rudimentary metrics.

Reply:
done